addons/account_edi/models/account_edi_document.py

```diff
@@ -178,6 +181,11 @@ def _postprocess_cancel_edi_results(documents, edi_result):
                 invoices.button_draft()
                 invoices.button_cancel()
 
+            if payment_ids_to_cancel:
+                payments = self.env['account.move'].browse(list(payment_ids_to_cancel))
+                payments.button_draft()
+                payments.button_cancel()
```

Comment:
@smetl You do not remember a reason why we did not include the payments for the cancellation in the first place?  (it might just be because it was less of a priority)

Reply:
@jco-odoo The payment flow is automatic. The payment is sent automatically when all invoices are paid. If you break a reconciliation, the payment is automatically cancelled I think. Here, we are just breaking the flow... I would like to avoid this huge bunch of code in a stable version..